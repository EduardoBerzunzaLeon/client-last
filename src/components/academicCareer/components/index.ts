export * from './StudentCard';
export * from './UserCreatorCard';
