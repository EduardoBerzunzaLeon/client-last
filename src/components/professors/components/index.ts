export * from './ButtonProfessorActive';
export * from './columns';
export * from './course';
export * from './ExcelButtonProfessors';
export * from './Header';
export * from './ProfessorDataForm';
export * from './ProfessorDialog';
