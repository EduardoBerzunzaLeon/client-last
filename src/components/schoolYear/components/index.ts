export * from './PhaseCard';
export * from './FileErrorsTable';
export * from './CurrentSubjectsForm';
export * from './FailedSubjectsForm';
export * from './ValidationPasswordForm';
export * from './SchoolYearStepForm';
