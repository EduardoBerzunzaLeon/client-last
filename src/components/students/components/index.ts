export * from './columns';
export * from './Header';
export * from './StudentDataForm';
export * from './StudentDialog';
export { ProfessorsHistoryDialog } from './professorsHistory';
