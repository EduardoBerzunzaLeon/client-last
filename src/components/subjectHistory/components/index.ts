export * from './SubjectsStudied';
export * from './CurrentSubjects';
export * from './UnstudySubjects';
export * from './columns';
export * from './PhaseDialog';
export * from './Header';
export * from './PhaseDataForm';
