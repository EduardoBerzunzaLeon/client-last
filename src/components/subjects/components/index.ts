export * from './Header';
export * from './columns';
export * from './SubjectDialog';
export * from './SubjectsList';
export * from './SubjectDetailDialog';
