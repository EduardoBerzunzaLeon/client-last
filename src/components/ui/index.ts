export * from './badge/Badge';
export * from './divider/Divider';
export * from './errorCard/ErrorCard';
export * from './excelButton/ExcelButton';
export * from './headerAdmin/HeaderAdmin';
export * from './skeleton';
export * from './skeletonDropdown/SkeletonDropdown';
export * from './slipButton/SlipButton';
export * from './spanDetail/SpanDetail';
export * from './spinner/Spinner';
export * from './spinnerRTK/SpinnerRTK';
