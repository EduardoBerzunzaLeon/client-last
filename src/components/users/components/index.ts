export * from './AdminPasswordForm';
export * from './ButtonUserBlocked';
export * from './columns';
export * from './Header';
export * from './UserDataForm';
export * from './UserDialog';
