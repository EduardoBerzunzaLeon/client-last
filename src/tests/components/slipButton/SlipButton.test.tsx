import { mount } from 'enzyme';

import { SlipButton } from '../../../components/ui';

describe('<SlipButton />', () => {
  test('should match to snapshot and disabled button when the loading is true', () => {
    const wrapper = mount(
      <SlipButton
        color="purple"
        icon="google"
        label="Google"
        loading
        onClick={() => {}}
      />,
    );
    const button = wrapper.find('button');
    expect(button.prop('disabled')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('should enabled button when the loading is false', () => {
    const mockFn = jest.fn();

    const wrapper = mount(
      <SlipButton
        color="purple"
        icon="google"
        label="Google"
        loading={false}
        onClick={mockFn}
      />,
    );

    const button = wrapper.find('button');
    button.simulate('click');
    expect(button.prop('disabled')).toBe(false);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
