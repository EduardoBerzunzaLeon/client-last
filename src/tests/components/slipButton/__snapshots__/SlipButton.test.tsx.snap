// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlipButton /> should match to snapshot and disabled button when the loading is true 1`] = `
<SlipButton
  color="purple"
  icon="google"
  label="Google"
  loading={true}
  onClick={[Function]}
>
  <div
    className="container-slip "
  >
    <ForwardRef
      className="button-slip purple p-0 mt-1 w-full"
      loading={true}
      onClick={[Function]}
    >
      <ButtonComponent
        badge={null}
        badgeClassName={null}
        className="button-slip purple p-0 mt-1 w-full"
        disabled={false}
        forwardRef={null}
        icon={null}
        iconPos="left"
        label={null}
        loading={true}
        loadingIcon="pi pi-spinner pi-spin"
        onClick={[Function]}
        tooltip={null}
        tooltipOptions={null}
      >
        <button
          className="p-button p-component button-slip purple p-0 mt-1 w-full p-button-icon-only p-disabled p-button-loading"
          disabled={true}
          onClick={[Function]}
        >
          <span
            className="p-button-icon p-c p-button-loading-icon pi pi-spinner pi-spin"
          />
          <i
            className="pi pi-google px-2"
          />
          <span
            className="px-3"
          >
            Google
          </span>
          <Ripple />
        </button>
      </ButtonComponent>
    </ForwardRef>
  </div>
</SlipButton>
`;
